frontend/src/components/VendorInfoNav.js

```diff
@@ -96,15 +98,18 @@ const VendorInfoNav = ({
                       />
                     </Link>
                   </div>
-        );
-                }) : <p className='empty-page-message'>Nincs megjeleníthető piac</p>
-      }
-            </div>
+                );
+              })
+            )
+: (
+              <p className='empty-page-message'>Nincs megjeleníthető piac</p>
+            )}
           </div>
+        </div>
       )
 : (
         <div className='profile-components'>
-          <Products products={products} />
+          {products.length > 0 && <Products products={products} />}
```

Comment:
Mert ha jól látom ez az egy sor a lényegi változás és ez rendben is van, a többi csak újraformázás.